dipy/reconst/qti.py

```diff
@@ -0,0 +1,652 @@
+    """
+    if V.shape[-2::] != (6, 1):
+        raise ValueError('The shape of the input array must be (..., 6, 1).')
+    C = 1 / np.sqrt(2)
+    T = np.array(([V[..., 0, 0], C * V[..., 5, 0], C * V[..., 4, 0]],
+                  [C * V[..., 5, 0], V[..., 1, 0], C * V[..., 3, 0]],
+                  [C * V[..., 4, 0], C * V[..., 3, 0], V[..., 2, 0]]))
+    T = np.moveaxis(T, (0, 1), (-2, -1))
+    return T
+
+
+def from_6x6_to_21x1(T):
+    """Convert symmetric 6 x 6 matrices into 21 x 1 vectors.
+
+    Parameters
+    ----------
+    T : numpy.ndarray
+        An array of size (..., 6, 6).
+
+    Returns
+    -------
+    V : numpy.ndarray
+        Converted vectors of size (..., 21, 1).
+    """
+    if T.shape[-2::] != (6, 6):
+        raise ValueError('The shape of the input array must be (..., 6, 6).')
+    if not np.all(np.isclose(T, np.swapaxes(T, -1, -2))):
+        warn('All input tensors are not symmetric.')
+    C = np.sqrt(2)
+    V = np.stack(([T[..., 0, 0], T[..., 1, 1], T[..., 2, 2],
+                   C * T[..., 1, 2], C * T[..., 0, 2], C * T[..., 0, 1],
+                   C * T[..., 0, 3], C * T[..., 0, 4], C * T[..., 0, 5],
+                   C * T[..., 1, 3], C * T[..., 1, 4], C * T[..., 1, 5],
+                   C * T[..., 2, 3], C * T[..., 2, 4], C * T[..., 2, 5],
+                   T[..., 3, 3], T[..., 4, 4], T[..., 5, 5],
+                   C * T[..., 3, 4], C * T[..., 4, 5], C * T[..., 5, 3]]),
+                 axis=-1)[..., np.newaxis]
+    return V
+
+
+def from_21x1_to_6x6(V):
+    """Convert 21 x 1 vectors into symmetric 3 x 3 matrices.
+
+    Parameters
+    ----------
+    V : numpy.ndarray
+        An array of size (..., 21, 1).
+
+    Returns
+    -------
+    T : numpy.ndarray
+        Converted matrices of size (..., 6, 6).
+    """
+    if V.shape[-2::] != (21, 1):
+        raise ValueError('The shape of the input array must be (..., 21, 1).')
+    C = 1 / np.sqrt(2)
+    T = np.array(
+        ([V[..., 0, 0], C * V[..., 5, 0], C * V[..., 4, 0],
+          C * V[..., 6, 0], C * V[..., 7, 0], C * V[..., 8, 0]],
+         [C * V[..., 5, 0], V[..., 1, 0], C * V[..., 3, 0],
+          C * V[..., 9, 0], C * V[..., 10, 0], C * V[..., 11, 0]],
+         [C * V[..., 4, 0], C * V[..., 3, 0], V[..., 2, 0],
+          C * V[..., 12, 0], C * V[..., 13, 0], C * V[..., 14, 0]],
+         [C * V[..., 6, 0], C * V[..., 9, 0], C * V[..., 12, 0],
+          V[..., 15, 0], C * V[..., 18, 0], C * V[..., 20, 0]],
+         [C * V[..., 7, 0], C * V[..., 10, 0], C * V[..., 13, 0],
+          C * V[..., 18, 0], V[..., 16, 0], C * V[..., 19, 0]],
+         [C * V[..., 8, 0], C * V[..., 11, 0], C * V[..., 14, 0],
+          C * V[..., 20, 0], C * V[..., 19, 0], V[..., 17, 0]]))
+    T = np.moveaxis(T, (0, 1), (-2, -1))
+    return T
+
+
+# These tensors are used in the calculation of QTI parameters
+e_iso = np.eye(3) / 3
+E_iso = np.eye(6) / 3
+E_bulk = np.matmul(from_3x3_to_6x1(e_iso), from_3x3_to_6x1(e_iso).T)
```

Comment:
Maybe this would improve readability?  

```suggestion
E_bulk = from_3x3_to_6x1(e_iso) @ from_3x3_to_6x1(e_iso).T
```

Reply:
There are a few other places where `np.matmul` is used, where I think readability would improve by using `@`